tests/test_io_dummy_param.rs

```diff
@@ -2,59 +2,100 @@
+        // 1. Generate RLWE ciphertext (a, b) for input k
+        // b = a * t - k * q/2 + e
+        let rlwe_encryption_sigma = 3.0;
+
+        // Generate random plaintext bits
+        let k = sampler_uniform.sample_poly(&params, &DistType::BitDist);
+
+        // Encrypt the plaintext
+        let (a, b, t) = encrypt_rlwe(&params, &sampler_uniform, rlwe_encryption_sigma, &k);
+
+        // decompose the polynomials a and b into bits
+        let a_bits = a.decompose(&params);
+        let b_bits = b.decompose(&params);
+
+        assert!(a_bits.len() == b_bits.len());
+        assert!(a_bits.len() == log_q);
+
+        // 2. Create a public circuit
+        // Input: a_bits[0], ..., a_bits[logq - 1], b_bits[0], ..., b_bits[logq - 1], x
+        // Output: a_bits[0] AND x, ..., a_bits[logq - 1] AND x, b_bits[0] AND x, ..., b_bits[logq -
+        // 1] AND x
+        // a_bits and b_bits are hardcoded inside the circuit
         let mut public_circuit = PolyCircuit::new();
-        {
-            let inputs = public_circuit.input(log_q + 1);
-            let mut outputs = vec![];
-            let eval_input = inputs[log_q];
-            for enc_input in inputs[0..log_q].iter() {
-                let muled = public_circuit.and_gate(*enc_input, eval_input);
-                outputs.push(muled);
-            }
-            public_circuit.output(outputs);
+        let x_id = public_circuit.input(1)[0];
+
+        let mut public_circuit_outputs = Vec::new();
+
+        let a_bits_vecs = a_bits.iter().map(|poly| poly.to_bool_vec()).collect::<Vec<_>>();
+        let b_bits_vecs = b_bits.iter().map(|poly| poly.to_bool_vec()).collect::<Vec<_>>();
+
+        for i in 0..a_bits_vecs.len() {
+            let a_ith_bits_const = public_circuit.const_bit_poly(&a_bits_vecs[i]);
+            let a_ith_bits_and_x = public_circuit.and_gate(a_ith_bits_const, x_id);
+            public_circuit_outputs.push(a_ith_bits_and_x);
         }
 
+        for i in 0..b_bits_vecs.len() {
+            let b_ith_bits_const = public_circuit.const_bit_poly(&b_bits_vecs[i]);
+            let b_ith_bits_and_x = public_circuit.and_gate(b_ith_bits_const, x_id);
+            public_circuit_outputs.push(b_ith_bits_and_x);
+        }
+
+        public_circuit.output(public_circuit_outputs);
+
+        // 3. Obfuscate the circuit
         let obf_params = ObfuscationParams {
             params: params.clone(),
             switched_modulus,
             input_size: 1,
             public_circuit: public_circuit.clone(),
             d: 3,
             encoding_sigma: 0.0,
-            hardcoded_key_sigma: 0.0,
             p_sigma: 0.0,
         };
 
-        let sampler_uniform = DCRTPolyUniformSampler::new();
-        let sampler_hash = DCRTPolyHashSampler::<Keccak256>::new([0; 32]);
-        let sampler_trapdoor = DCRTPolyTrapdoorSampler::new(&params, SIGMA);
         let mut rng = rand::rng();
+
+        let t_mat = DCRTPolyMatrix::from_poly_vec_column(&params, vec![t.clone()]);
```

Comment:
As I mentioned above, this is defined as `t_bar` in the paper.
`t = (t_bar, -1)`